### ISSUE TITLE
Export Orders Status for all states and process the order on magento appropriately.

### ISSUE THREAD

**Issue by shalabhaggarwal** [openlabs/magento_integration#38]
Right now the order status export works only for cancelled orders. This should work with all the states between openerp and magento.

**Comment by janpfischer** [comment]
That would be so nice! I can help out if you want. Need that badly.

**Comment by shalabhaggarwal** [comment]
Hi @b1rdhous3 

Can you explain the way you would prefer this feature to be implemented? We are gathering feedbacks from users and developers in community to build this feature in the best possible way.

Thanks!

**Comment by janpfischer** [comment]
Hey @shalabhaggarwal 

for example if you create an invoice there should be a status update to magento or if you create a shipment. Not only provide shipment information to magento. Update a status, too.

Or if the OpenERP Workflow is "Done" mark the magento order "Done".

A good example is the Magento OpenERP Connection called "AiO-Connect" Google for that.

**Comment by capitaine** [comment]
Magento most usual status are: 
- processing (orders have an online payment (credit card or paypal) and magento have already created an invoice) 
  >> they should be considered as Orders in openERP because they can be delivered. What do we do with the magento's invoice (see #107)
- pending (orders have an offset payment : check or bank transfer; they are not yet invoiced) 
  >> They should be considered as Quotes in openERP or much better a checkbox choice to consider them like Quotes or Orders (because if they are Quotes, I think they are not taken in consideration in the restocking process, and some people may wish to go on with restocking. This depends if you trust your customers that they will send you the money or not, and this depends on the history/experience of your shop)
- pending payment: in my case this is the status when customers abandon the online payment process.
  >> They should not be imported in openERP. Who needs them ? To do what ? Or we should have the choice to import them or not
- canceled: Why do we need them in openERP. In don't see any use of these recordings, especially when you know that Magento creates an order (which in reality is just a quote) just after validating an order but before paying it. So many customers create several canceled-orders before they achieve a valid order. A lot of reasons for that: they don't understand the Secure 3D (Visa, Mastercard "new" security), they just want to verify a last time their order and click the browser's "back" button, their wife irrupted in their room just while they were typing the credit card number, and so on... So why import these orders/quotes in openERP ?
  >> They should not be imported in openERP or we should have the choice to do it or not, because I also think that all shops don't have the same customer's behavior that I described.
- Complete: order has been delivered or at least posted. Is it useful to have it in the orders list ? Perhaps import it as an invoice (but the invoice may be different from the order). No clear idea yet
- Closed: why is it closed ? in my shop it is when an invoiced order (processing or processed, not sure) is refunded through a credit note. Not yet a clear idea how/where it should be imported. Should we also  import the associated credit note ?

**Comment by sharoonthomas** [comment]
I feel that the order status with safe default implement almost the same functionality that you have mentioned. In addition it also gives you the option to explicitly avoid importing orders in status' which it should not. 

@shalabhaggarwal Can we consider this implemented ?

Please feel free to send a pull request if the default processing for the order states are not appropriate

**Comment by shalabhaggarwal** [comment]
@sharoonthomas Yes, the current implementation provides complete flexibility to the user to choose which order state to import orders in and how they should be mapped in openerp.

If there is something specific to any order state processing, that should be handled in a separate issue.
